src/olympia/addons/admin.py

```diff
@@ -83,6 +85,29 @@ def status_with_admin_manage_link(self, obj):
                            link, obj.get_status_display())
     status_with_admin_manage_link.short_description = _(u'Status')
 
+    def change_view(self, request, object_id, form_url='', extra_context=None):
+        lookup_field = Addon.get_lookup_field(object_id)
```

Comment:
We need something more DRY here... Can you extract the common code with `addon_view` decorator and re-use that ?

Reply:
I decided to duplicate this code because here we are redicrecting to the AMO id (vs the slug). I think if we want to keep the redirects in place, having those distinct implementations make sense because we probably don't want this one re-used on a public page. Let me know what you think.